packages/@magic-sdk/provider/src/modules/auth.ts

```diff
@@ -141,6 +143,29 @@ export class AuthModule extends BaseModule {
       this.sdk.testMode ? MagicPayloadMethod.UpdateEmailTestMode : MagicPayloadMethod.UpdateEmail,
       [{ email, showUI }],
     );
-    return this.request<string | null, UpdateEmailEvents>(requestPayload);
+
+    const handle = this.request<string | null, UpdateEmailEventHandlers>(requestPayload);
+
+    if (!showUI) {
```

Comment:
why could we not emit these events regardless of whether showUI is true or not?

Reply:
This is mainly for performance concerns. Adding too many listeners would consume extra resources in the device.